lms/djangoapps/courseware/rules.py

```diff
@@ -24,3 +28,21 @@ def is_track_ok_for_exam(user, exam):
 # proctored experience
 can_take_proctored_exam = is_track_ok_for_exam
 rules.set_perm('edx_proctoring.can_take_proctored_exam', is_track_ok_for_exam)
+
+
+class HasAccessRule(Rule):
+    """
+    A rule that calls `has_access` to determine whether it passes
+    """
+    def __init__(self, action):
+        self.action = action
+
+    def check(self, user, instance=None):
+        return has_access(user, self.action, instance)
+
+    def query(self, user):
```

Comment:
Is this the only difference from the original PR?

Do we want to add any sort of test to make sure we don't break the admin view again?

Reply:
Yeah, this is the only change.

It does seem like it would be worthwhile to make sure the admin view works. Not sure where that test would live, though. I'll think on it.